src/vulkan/device.cc

```diff
@@ -578,7 +621,7 @@ Result Device::Initialize(
   const bool needs_subgroup_size_control =
       std::find(required_features.begin(), required_features.end(),
                 kSubgroupSizeControl) != required_features.end();
-  if (needs_subgroup_size_control && use_physical_device_features_2) {
+  if (needs_subgroup_size_control) {
     PFN_vkGetPhysicalDeviceProperties2KHR vkGetPhysicalDeviceProperties2KHR =
```

Comment:
Can you explain why you were able to remove this?

Reply:
Previously the usage of vkGetPhysicalDeviceFeatures2 was determined by an extension. But now that it's promoted to core I don't think we should require the scripts to enable the extension. And since we check for both feature structs we don't really need to check this. After all we don't call the vkGetPhysicalDeviceFeatures2 here.

Reply:
Promoted to core? But in Vulkan 1 it will still be an extension and so `getInstanceProcAddr` might return null?

Reply:
You're right. I need to restore the check for the extension and also query the Vulkan API version and use the core function in case of Vulkan >= 1.1.

Reply:
I think you might be able to always use the extension, even if it is core in the Vulkan being used.

Reply:
Which means the script needs to have that extension enabled if it wants to be compatible with Vulkan 1.0. 

Does the spec somewhere mention you have to still support the extensions after being promoted to core? If not, it might be safest to just check the version as it's quite easy to do.

Reply:
The notice of promotion is like this: "All functionality in this extension is included in core Vulkan 1.1, with the KHR suffix omitted. The original type, enum and command names are still available as aliases of the core functionality."

I think Amber should (and already does, I think) implicitly add required instance extensions (like `VK_KHR_get_physical_device_properties2`) when it knows they will be required due to other required extensions.

Reply:
I've now restored the checking of that extension.